File: C Sharp/Object Validator Test/TestObject.cs
using System.Globalization;

class TestObject(string? testString, int testInt, bool testBool, TestObject2?[]? testArray)
{
    private readonly string? testString = testString;
    private readonly int testInt = testInt;
    private readonly bool testBool = testBool;
    private readonly TestObject2?[]? testArray = testArray;

    public string? TestString => testString;
    public int TestInt => testInt;
    public bool TestBool => testBool;
    public TestObject2?[]? TestArray => testArray;
}

class TestObject2(DateTime testDateTime)
{
    private readonly DateTime testDateTime = testDateTime;

    public override string ToString()
    {
        return testDateTime.ToString("s", CultureInfo.InvariantCulture);
    }
}

File: Java/Object Validator/lib/src/test/java/com/quicksilver/objectvalidator/TestObject.java
package com.quicksilver.objectvalidator;

public class TestObject {
    private final String testString;
    private final int testInt;
    private final boolean testBool;
    private final TestObject2[] testArray;

    public TestObject(String testString, int testInt, boolean testBool, TestObject2[] testArray) {
        this.testString = testString;
        this.testInt = testInt;
        this.testBool = testBool;
        this.testArray = testArray;
    }

    public String getTestString() {
        return testString;
    }

    public int getTestInt() {
        return testInt;
    }

    public boolean isTestBool() {
        return testBool;
    }

    public TestObject2[] getTestArray() {
        return testArray;
    }
}